src/clippy-tracing/src/main.rs

```diff
@@ -0,0 +1,455 @@
+// Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#![warn(clippy::pedantic)]
+
+//! A tool to add, remove and check for `tracing::instrument` in large projects where it is
+//! infeasible to manually add it to thousands of functions.
+
+use std::collections::HashMap;
+use std::error::Error;
+use std::fmt;
+use std::fs::OpenOptions;
+use std::io::{Read, Write};
+use std::path::PathBuf;
+use std::process::ExitCode;
+
+use clap::{Parser, ValueEnum};
+use syn::spanned::Spanned;
+use syn::visit::Visit;
+use walkdir::WalkDir;
+
+/// The command line arguments for the application.
+#[derive(Parser)]
+struct CommandLineArgs {
+    /// The action to take.
+    #[arg(long)]
+    action: Action,
+    /// The path to look in.
+    #[arg(long)]
+    path: Option<PathBuf>,
+    /// When adding instrumentation use a custom suffix e.g.
+    /// `tracing::instrument` vs `my::custom::suffix::instrument`.
```

Comment:
instead of `tracing::instrument` and `my::custom::suffix::instrument` can we simply say that a valid example of suffix is `--suffix my::custom::suffix::`.

Reply:
Updated.